src/test/java/giis/eshopcontainers/e2e/functional/common/BaseLoggedTest.java

```diff
@@ -0,0 +1,181 @@
+
+import giis.eshopcontainers.e2e.functional.common.exceptions.ElementNotFoundException;
+import giis.eshopcontainers.e2e.functional.model.EShopUser;
+import giis.eshopcontainers.e2e.functional.utils.Click;
+import giis.eshopcontainers.e2e.functional.utils.Navigation;
+import io.github.bonigarcia.wdm.managers.ChromeDriverManager;
+import io.github.bonigarcia.wdm.managers.FirefoxDriverManager;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.logging.LogEntries;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Date;
+import java.util.Properties;
+
+import static org.openqa.selenium.logging.LogType.BROWSER;
+
+public class BaseLoggedTest {
+    public static final Logger log = LoggerFactory.getLogger(BaseLoggedTest.class);
+    public static final String CHROME = "chrome";
+    protected static final Class<? extends WebDriver> chrome = ChromeDriver.class;
+    protected static final Class<? extends WebDriver> firefox = FirefoxDriver.class;
+    public static String CLIENT_BROWSER;
+    public String testName;
+    public static final String LOCALHOST = "http://156.35.119.57:5100";
+    protected static String SUT_URL;
+    protected static Properties properties;
+    protected static String TJOB_NAME = "DEFAULT_TJOB";
+    private static EShopUser logged_user;
+
+    protected BrowserUser user;
+
+    @BeforeAll()
+    static void setupAll() { //28 lines
+        properties = new Properties();
+        logged_user = new EShopUser("alice", "Pass123$");
+        try {
+            // load a properties file for reading
+            properties.load(Files.newInputStream(Paths.get("src/test/resources/inputs/test.properties")));
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        if (System.getProperty("SUT_URL") == null) {
+            // Outside CI
+            ChromeDriverManager.getInstance(chrome).setup();
+            FirefoxDriverManager.getInstance(firefox).setup();
+        }
+        if (System.getProperty("SUT_URL") != null) {
+
+            SUT_URL = "http://" + System.getProperty("SUT_URL") + ":" + System.getProperty("SUT_PORT") + "/";
+            log.info("Configuring CI system SUT: " + SUT_URL);
+        } else {
+            SUT_URL = LOCALHOST;
+
+        }
+        CLIENT_BROWSER = System.getenv("CLIENT_BROWSER");
+        if ((CLIENT_BROWSER == null)) {
+            CLIENT_BROWSER = CHROME;
+        }
+
+    }
```

Comment:
> I'm not sure whether using environment variables is better approach thatn using a properties file to configure the test. Using environmente variables leads to more code by checking nulls and with defaults hardwritten in the source code instead of externalized in a properties file.

I am utilizing environment variables because in the event of parallel instances of the System Under Test (SUT), such as with four different eShopContainers, various URLs for the webmvc can be dynamically passed through these variables. For instance, the first one could be '[http://webmvc-tjobA:80](http://webmvc-tjoba/)', the second '[http://webmvc-tjobB:80](http://webmvc-tjobb/)', and so on.

Reply:
ok, then the approach is use properties that are "global" and using env vars when they are TJob specific, right?

Reply:
Yes!